sonaranalyzer-dotnet/src/SonarAnalyzer.CSharp/CBDE/CbdeHandler.cs

```diff
@@ -35,6 +35,43 @@
 
 namespace SonarAnalyzer.Rules.CSharp
 {
+    public class ThreadCpuStopWatch
+    {
+        private double totalMsStart;
+        private readonly ProcessThread currentProcessThread;
+        public long ElapsedMilliseconds
+        {
+            get { return (long)(currentProcessThread.TotalProcessorTime.TotalMilliseconds - totalMsStart); }
+        }
+        public static ProcessThread GetCurrentProcessThread()
+        {
+            var currentId = AppDomain.GetCurrentThreadId();
+            foreach (ProcessThread p in Process.GetCurrentProcess().Threads)
```

Comment:
```suggestion
            return Process.GetCurrentProcess().Threads.Single(p => p.Id == currentId);
```

Reply:
not a generic collection, so it is not possible